### ISSUE TITLE
Can't install Royalty armor glands on the "body" bodypart

### ISSUE THREAD

**Issue by SamaelGray** [SineSwiper/RimWorld-XenobionicPatcher#19]
Royalty armor glands being toughskin, armor skin and stoneskin glands are installable on the torso which can be done for both animals and humanoids thanks to Xenobionic Patcher.

Problem is certain animals lack the "torso" bodypart and instead have "body" as the main bodypart. Can XP be coded in a way that it detects and considers "body" and "torso" as the same bodypart when installing implants on the torso bodypart?

**Comment by SineSwiper** [comment]
[This is already setup here.](https://github.com/SineSwiper/RimWorld-XenobionicPatcher/blob/master/Source/DefInjectors.cs#L138)  Can you specify which animal is having the issue?

**Comment by SamaelGray** [comment]
Any animal that has "body" instead of "torso"?
Vanilla and modded ones, there is no difference. Unless there is a setting in the mod that I'm missing which I highly doubt.
XB is loaded near the bottom of my modlist btw.

**Comment by SamaelGray** [comment]
XB settings, surgery menu with and without CE is the same for animals and an animal example that has "body" instead of "torso"
![image](https://user-images.githubusercontent.com/56392968/180836018-5f7ab31c-035d-4e8b-b4fe-1c5d69f8f5a1.png)

A donkey:
![image](https://user-images.githubusercontent.com/56392968/180835768-d15d4075-b35a-4ada-a79d-2ea2c59d77da.png)
A normal human pawn:
![image](https://user-images.githubusercontent.com/56392968/180835785-636854fb-ccf1-429e-9c32-bba67a4960c0.png)

**Comment by SineSwiper** [comment]
Does it work if `Animal Surgeries → Humanlikes` is also checked?  There's a special pass if all four of the "fleshlikes" are checked.  Also, what does the log say on load when you check `More Debug Info`?

**Comment by SamaelGray** [comment]
The log at startup after enabling `Animal Surgeries → Humanlikes`:
https://gist.github.com/3a9a0d3f27facbc3e122e21019467d80

Enabling `Animal Surgeries → Humanlikes` did not change the result.
Enabling all the settings did have change the results either.

The logs after enabling all the options:
https://gist.github.com/2de54f09b89fedaf91fd1ca5ac72c937

Tested on a Warg and a normal human pawn. There must be some check in the code that is not triggering, not functioning properly or something, I'm not a code savvy so I won't be much of a help there.
If you have a discord ID, we can talk over there more easily, mine is `Sam Gray#0772`.

**Comment by SineSwiper** [comment]
```
[XenobionicPatcher] Injecting humanlike surgical recipes into animals
[XenobionicPatcher]     PawnSurgeriesByBioType cache: took 0.0000s; 1/18 keys/recipes
[XenobionicPatcher]     RaceBodyParts cache: took 0.0000s; 402 BPRs
[XenobionicPatcher]     DoesPawnHaveSurgery + BodyPart caches: took 0.0010s; 1,242 + 1,197 strings
[XenobionicPatcher]     Static part loop: took 0.0070s; 38/42 PartToPartMapper keys/BPDs
[XenobionicPatcher]     Part-to-part mapping: took 0.0000s; 50/71 PartToPartMapper keys/BPDs
[XenobionicPatcher]     Surgery-to-part mapping: took 0.0000s; 50/71 PartToPartMapper keys/BPDs
[XenobionicPatcher]     Empty list cleanup: took 0.0000s; 50/71 PartToPartMapper keys/BPDs
[XenobionicPatcher]     Add new body parts to surgeries: took 0.0000s; 9 additions
[XenobionicPatcher]     Add new surgeries to pawns: took 0.0030s; 3,948 additions
[XenobionicPatcher] Injected humanlike surgical recipes into animals (took 0.0130s; 5,451 combinations)
```

Well, it's doing something.  Almost 4K additions, but they aren't showing up on the menu above.  It must be some sort of mod conflict.

Is it Recipe Inheritance?  Combat Extended?  (Though, I would expect more complaints.)  Multiplayer?  Might want to just start removing mods from the list, reload the game, and see if you get better surgery results.  Obviously, red errors and missing objects are fair game when you remove mods, but you don't have to save the game in that state.  Use RimPy or another mod manager to keep your original mod list.

**Comment by SineSwiper** [comment]
Actually, I'm able to duplicate the issue myself.  I'm troubleshooting it now...

**Comment by SineSwiper** [comment]
This issue has been fixed in v1.2.5.
